# Exploring Apple Appstore and Google Playstore

The goal of this project is to explore data of the Apple Appstore and Google playstore to determine what types of apps are the most popular across both platforms. This is a guided project with Dataquest.io and we will only be using Python to complete this analysis.

## The Datasets

Both the Appstore and Playstore contain millions of apps for users to choose from. To save time with this project we will be using the following datasets from Kaggle:

__[Google Play Store](https://www.kaggle.com/datasets/lava18/google-play-store-apps)__

__[Apple AppStore](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps)__


## Reading in the Data

In [5]:
from csv import reader

### We will read in the data using the csv module reader function. 
### We will not be including the header row of either dataset.
### Instead the header rows will be saved to a unique variable for later use.

### Google Play Apps ###

opened_file = open('googleplaystore.csv', encoding = "utf-8")
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### App Store Apps ###
opened_file = open('AppleStore.csv', encoding = "utf-8")
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

Now, let's create a function that can take in the datasets and return exploratory information on each dataset. This function will take in the dataset, a beginning and ending row and a boolean variable so we can get the number of rows and columns from each dataset.

In [6]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    #This for loop will display each row requested
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

Now let's test our function with each dataset. 

In [7]:
print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [8]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 7197
Number of columns: 17


From this initial exploration we can determine which columns will be useful for our analysis.

## Deleting Wrong Data

From the discussion session of this dataset on Kaggle, we saw that there was inoccrect data in our android dataset. The following row has missing data in the 'Genre' category and is missing the 'Category' column altogether. So we will delete that row from our dataset.

In [9]:
print(android[10472])  # incorrect row
print('\n')
print(android_header)  # header
print('\n')
print(android[0])  

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In [10]:
print(len(android))
del android[10472]  # don't run this more than once
print(len(android))

10841
10840


## Finding Duplicate Data - Part 1

Now let's find duplicate data in our datasets. We will use a for loop that will loop through the dataset and append duplicates to a list and display a selection of duplicated apps. 

In [11]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicate apps: ', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps: ', duplicate_apps[:15])

Number of duplicate apps:  1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [12]:
### Let's look at an example of a duplicated app and try to determine a criteria for duplicate deletion

for app in android:
    name = app[0]
    if name == 'Google Ads':
        print(app)

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


We see from the above example that the difference between the apps are in the fourth column that lists the number of reviews of the app. We will assume that the higher the number of reviews the more recent the entry and use that as the criteria for determining which apps to delete. 

## Removing the Duplicate Data - Part 2

Using a dictionary, we will loop through the data set and convert our reviews data to a float. Then we will use the name of the app as a key and assign to it the highest value of the reviews column.

In [13]:
reviews_max = {}

for app in android:
    name = app[0]
    reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < reviews:
        reviews_max[name] = reviews
        
    elif name not in reviews_max:
        reviews_max[name] = reviews 

In [14]:
#Let's ensure we got the correct amount of data in our dictionary 

print("Expected length: ", len(android) - 1181)
print("Actual length: ", len(reviews_max))

Expected length:  9659
Actual length:  9659


Now we can loop through our dataset and append our cleaned data to a list that contains only one entry for each app. 

In [15]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    reviews = float(app[3])
    #This will add the apps with the highest review to the clean list.
    if (reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [16]:
#Let's explore the data to ensure our for loop worked as intended

explore_data(android_clean, 0, 4, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 9659
Number of columns: 13


Now that we have cleaned up the duplicates we can focus on removing non-English apps from our datasets since those will not be relevant to our audience.

## Removing Non-English Apps

Let's begin by printing out some examples of non-English apps from each dataset.

In [17]:
print(ios[813][1])
print(ios[6731][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])

436672029
1144164707


中国語 AQリスニング
لعبة تقدر تربح DZ


We can begin by creating a function that will take in the name of the app as a string that will return a boolean variable determining if the name is English or not. Since English ASCII's number range goes up to 127, we will use that to determine if an app is in English or not.  

In [18]:
def english_app(string):
    
    for char in string:
        if ord(char) > 127:
            return False

    return True

print(english_app('Instagram'))
print(english_app('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_app('Docs To Go™ Free Office Suite'))
print(english_app('Instachat 😜'))

True
False
False
False


We can see that we ran into an issue when the name using special characters such as emojis. Let's see where they fall in the ASCII system.

In [19]:
print(ord('™'))
print(ord('😜'))

8482
128540


We will now rework our function to account for this. We will include a boolean statement that will determine whether or not the app is in English if the app has less than three special characters. 

In [20]:
def english_app(string):
    ascii_app = 0
    
    for char in string:
        if ord(char) > 127:
            ascii_app += 1
            
    if ascii_app > 3:
        return False
    else:
        return True
    
print(english_app('Docs To Go™ Free Office Suite'))
print(english_app('Instachat 😜'))

True
True


Now we can begin to create new datasets that include only the English apps. We will loop through each dataset and append only the English apps to our new lists using our above function.

In [21]:
ios_english = []
android_english = []

for app in ios:
    name = app[1]
    if english_app(name):
        ios_english.append(app)
        
for app in android_clean:
    name = app[0]
    if english_app(name):
        android_english.append(app)
        
#Let's explore the data to ensure it worked properly
        
explore_data(ios_english, 0, 3, True)
print('\n')
explore_data(android_english, 0, 3, True)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 7197
Number of columns: 17


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', '

## Isolating the Free Apps

The final part of our data cleaning process will be isolating only the free apps from our datasets. We will loop through each dataset and append only the apps that are priced at $0.

In [22]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)

#Let's see how many apps we have left in each dataset.        
        
print(len(ios_final))
print(len(android_final))

0
8864


## Most Common Apps by Genre

Now that we have our datasets cleaned we can begin to determine what type of apps have the most users. We want the app to be popular on both platforms because ideally it would be available for both Android and iPhone users. To do this we will divide the apps by genre and determine from there which apps are downloaded the most. 

We will begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

### Building the Functions

We will build two functions. The first function will display the frequency of each Genre/Category in a column shown as a percent.

The second function will take the percentage and display it along side the category of the app. 

In [23]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Now that we have the functions built, lets test them on our datasets.

In [24]:
display_table(ios_final, -5)

The table gives the impression that ios apps are primarily for fun/entertainment. It seems like other apps for productivity or education are not as popular on ios. Although entertainment apps make up more than half of the apps on ios, it does not indicate actualy usage by users.

Now let's look at the Google Play Store 'Category' column.

In [25]:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

It would appear that the Google Play Store has many more apps for productivity and tools for daily use. We know from other completed projects that the Family category are games for young children. So like ios entertainment apps make up a large portion of the apps, however it is not a large a portion as ios.

In [26]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

There isn't a clear explanantion of the difference between the 'Category' and 'Genre' columns for the Google Play Store. Since we're looking at the big picture we will only use the 'Category' column going forward as we look at the most popular apps for each store.

We know that entertainment apps make up a large portion of the apps for each store, but the Google Play Store seems to be more directed for tools and productivy than the iOS store.

Now lets look at the most popular apps.

## Most Popular Apps by Genre - App Store

The ideal way to find the most popular apps are to look at the install rate of an app. Since we do not have that number for the App Store, we will go based on the number of ratings an app has. 

We will loop through each of the categories and determine which of the categories has the highest number of ratings. 

In [27]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

The top three here are Navigation, Reference and Social Networking. Let's look at each of these top three and find out which apps are responsible for such high ratings.

In [28]:
# loop through the Navigation apps and print name and number of ratings
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) 

This may not be a good category to use to find our final answer because the results are heavily influenced by only two apps, Waze and Google Maps, which are the top two Navigation apps.

In [29]:
for app in ios_final:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5]) 

We see a similar thing here with the Social Networking apps. The most well-known popular apps have the highest ratings, and those after that are communication apps.

In [30]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5]) 

Finally, we see that the Reference section is heavily influenced by the Bible and Dictionary Apps. Although this seems like a niche category, it does provide some potential. We can take another open source popular book and create an app that has features that people find useful..

Now let's analyze the Google Play Store installs.

## Most Popular Apps by Genre - Google Play Store

Unlike the App Store, we have the install data of the Google Play Store, so we can use that to determine an apps popularity. 

In [31]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


As we can see these numbers are not precise, so we will convert the strings to floats and consider any app under the category has that many installs. Apps in 100,000 has 100,000 installs and so on. 

So let's convert the installs to floats. 

In [32]:
categories_android = freq_table(android_final, 1) #Save the frequency table of the Genre 

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        #Removes commas and plus sign, converts to float then gets the avg for each genre
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ":", avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Communication apps have the highest average of installs at 38,456,119. In the following code we will see that this number is highly influenced by the most popular communicating apps such as WhatsApp, Facebook Messenger, Skype, etc.:

In [33]:
#loops through dataset and pulls all apps with the respective install amount
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

We already saw the affect that the most popular apps have on the amount of reviews in the App Store, and it looks like we have something similar going on here in the Google Play Store. These apps are extremely popular among users and pose a threat if we wanted to create an app to compete against them. 

We want our app to be popular and have very little competition in the app store to increase installs. Our App Store data saw that the Reference category had some potential if we created an app for a popular book. 

Let's look at the Book & Reference section of the Google Play Store.

In [34]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [35]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


There are only a few highly popular apps in this Genre, so there is some potential to create an app that can become just as popular. Let's take a look at some apps that fall between 1million and 50million installs.

In [36]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

We see here that this is mainly full of reference material such as dictionaries and translation resources, as well as ebook readers. 

There are a number of apps that are centered around reading the Quran, so perhaps there is some potential there. Perhaps we can do something similar with another popular book and add in features that people find useful within the app. Such as an installed dictionary and not take feature so users do not have to leave our app while reading. We can also have an audio version and a forum where fans of the book and discuss the book. This shows the most potential for popularity with little competition in both markets.

## Conclusion

In this project we analyzed data about the Apple App Store and Google Play Store to find an app idea we can use that would be popular in both markets. 

We concluded that creating an app on a popular book could potentially be popular among both platforms. There is little competition, but we would need to add additional features beyond the book. We should include an audio version, forums for discussion among fans, and other relevant information fans of the book would appreciate. 